cpp/src/arrow/status.h

```diff
@@ -207,23 +218,68 @@ class ARROW_EXPORT Status {
   // a `State` structure containing the error code and message(s)
   State* state_;
 
-  void CopyFrom(const State* s);
+  void CopyFrom(const Status& s);
+  void MoveFrom(Status& s);
```

Comment:
This struck me as a tad odd (vs. `Status&& s`); I guess it doesn't matter too much because this is an internal API, so I think it's OK as is